tracer/src/Datadog.Trace.Tools.Runner/Gac/AdministratorHelper.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="AdministratorHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Security.Principal;
+
+namespace Datadog.Trace.Tools.Runner.Gac;
+
+#if NET5_0_OR_GREATER
+[SupportedOSPlatform("windows")]
+#endif
+internal static class AdministratorHelper
+{
+    private static bool? _isElevated;
+
+    public static bool IsElevated
+    {
+        get
+        {
+            _isElevated ??= new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator);
+            return _isElevated.Value;
+        }
+    }
+
+    public static void EnsureIsElevated()
+    {
+        if (!IsElevated)
+        {
+            var commandLineArguments = Environment.GetCommandLineArgs();
+#if NET6_0_OR_GREATER
+            var processPath = Environment.ProcessPath ?? commandLineArguments[0];
+#else
+            var processPath = commandLineArguments[0];
+#endif
+
+            var processInfo = new ProcessStartInfo(processPath)
+            {
+                Verb = "runas",
+                UseShellExecute = true,
+                CreateNoWindow = true
+            };
+
+            foreach (var arg in commandLineArguments.Skip(1))
+            {
+                processInfo.ArgumentList.Add(arg);
+            }
+
+            var process = Process.Start(processInfo);
+            process?.WaitForExit();
+            Console.WriteLine("Returned: {0}", process?.ExitCode);
+            Environment.Exit(process?.ExitCode ?? 0);
```

Comment:
Just personal preference, but I'd rather use `if (process is null)` instead of null-coalescing three times. We should probably also check for `null` in case it fails?

Maybe something like:

```suggestion
            if (process is null)
			{
			    // TODO: display error message
				// Environment.Exit(<not zero?>);
			}
			else
			{
				process.WaitForExit();
				Console.WriteLine("Returned: {0}", process.ExitCode);
				Environment.Exit(process.ExitCode);
			}
```

Reply:
I'll do this in a different PR just for the sake of not waiting again for CI.